src/DDTrace/OpenTelemetry/SpanBuilder.php

```diff
@@ -0,0 +1,212 @@
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function setSpanKind(int $spanKind): SpanBuilderInterface
+    {
+        $this->spanKind = $spanKind;
+
+        switch ($spanKind) {
+            case API\SpanKind::KIND_CLIENT:
+                $this->setAttribute(Tag::SPAN_KIND, Tag::SPAN_KIND_VALUE_CLIENT);
+                break;
+            case API\SpanKind::KIND_SERVER:
+                $this->setAttribute(Tag::SPAN_KIND, Tag::SPAN_KIND_VALUE_SERVER);
+                break;
+            case API\SpanKind::KIND_PRODUCER:
+                $this->setAttribute(Tag::SPAN_KIND, Tag::SPAN_KIND_VALUE_PRODUCER);
+                break;
+            case API\SpanKind::KIND_CONSUMER:
+                $this->setAttribute(Tag::SPAN_KIND, Tag::SPAN_KIND_VALUE_CONSUMER);
+                break;
+            case API\SpanKind::KIND_INTERNAL:
+                $this->setAttribute(Tag::SPAN_KIND, Tag::SPAN_KIND_VALUE_INTERNAL);
+                break;
+            default:
+                break;
+        }
+
+        return $this;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function startSpan(): SpanInterface
+    {
+        $parentContext = Context::resolve($this->parentContext);
+        $parentSpan = Span::fromContext($parentContext);
+        $parentSpanContext = $parentSpan->getContext();
+
+        $span = \DDTrace\start_trace_span($this->startEpochNanos);
+        $traceId = $parentSpanContext->isValid() ? $parentSpanContext->getTraceId() : \DDTrace\root_span()->traceId;
+
+        $samplingResult = $this
+            ->tracerSharedState
+            ->getSampler()
+            ->shouldSample(
+                $parentContext,
+                $traceId,
+                $this->spanName,
+                $this->spanKind,
+                $this->attributesBuilder->build(),
+                $this->links,
+            );
+        $samplingDecision = $samplingResult->getDecision();
+        $sampled = SamplingResult::RECORD_AND_SAMPLE === $samplingDecision;
+        $samplingResultTraceState = $samplingResult->getTraceState();
+
+        if ($parentSpanContext->isValid()) {
+            // Traceparent: {2:version}-{32:hex trace id}-{16:hex parent id}-{2:trace_flags}, version always being '00'
+            // Since parentSpanContext is valid, the trace identifiers are guaranteed to be in hexadecimal format
+            $parentId = $parentSpanContext->getSpanId();
+            $traceFlags = $sampled ? '01' : '00';
+            $traceParent = "00-$traceId-$parentId-$traceFlags";
+            \DDTrace\consume_distributed_tracing_headers([
+                'traceparent' => $traceParent,
+                'tracestate' => (string) $samplingResultTraceState, // __toString() is implemented in TraceState
+            ]);
+        } else {
+            \DDTrace\consume_distributed_tracing_headers([
+                'tracestate' => (string) $samplingResultTraceState, // __toString() is implemented in TraceState
+            ]);
```

Comment:
Does this even have any effect? If I read ddtrace_read_distributed_tracing_ids correctly, it does nothing without traceparent.

If you want to update the tracestate only, you should remove any `dd=` part and directly assign to RootSpanData->tracestate.

Also, if we implement my suggestion from another comment about decorating TraceStateInterface instead, this logic can probably just be skipped if we know it's just the unmodified tracestate of the parent.

Reply:
> Are we missing a test for this scenario?

I changed a bit that section (with an otherwise failing scenario as well)